nilearn/_utils/tests/test_helpers.py

```diff
@@ -23,6 +30,74 @@ def _mock_args_for_testing_replace_parameter():
     return mock_kwargs_with_deprecated_params_used, replacement_params
 
 
+@pytest.mark.skipif(
+    is_matplotlib_installed(),
+    reason="Test requires matplotlib not to be installed.",
+)
+def test_should_raise_warning_if_mpl_not_installed():
+    with (
+            pytest.warns(UserWarning, match="matplotlib is not installed"),
+            pytest.raises(
+                ModuleNotFoundError, match="No module named 'matplotlib'"
+            ),
+    ):
+        _set_mpl_backend(WARNING)
+
+
+@pytest.mark.skipif(
+    not is_matplotlib_installed(),
+    reason="Test requires matplotlib to be installed.",
+)
+@patch("matplotlib.use")
+@patch("matplotlib.get_backend", side_effect=["backend_1", "backend_2"])
+def test_should_raise_warning_if_backend_changes(*_):
+    # The backend values returned by matplotlib.get_backend are different.
+    # Warning should be raised to inform user of the backend switch.
+    with pytest.warns(UserWarning, match="Backend changed to backend_2..."):
+        _set_mpl_backend(WARNING)
+
+
+@pytest.mark.skipif(
+    not is_matplotlib_installed(),
+    reason="Test requires matplotlib to be installed.",
+)
+@patch("matplotlib.use")
+@patch("matplotlib.get_backend", side_effect=["backend_1", "backend_1"])
+def test_should_not_raise_warning_if_backend_is_not_changed(*_):
+    # The backend values returned by matplotlib.get_backend are identical.
+    # Warning should not be raised.
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        _set_mpl_backend(WARNING)
+
+
+@pytest.mark.skipif(
+    not is_matplotlib_installed(),
+    reason="Test requires matplotlib to be installed.",
+)
+@patch(
+    "matplotlib.use", side_effect=[Exception("Failed to switch backend"), True]
+)
+def test_should_switch_to_agg_backend_if_current_backend_fails(use_mock):
+    # First call to `matplotlib.use` raises an exception, hence the default Agg
+    # backend should be triggered
+    _set_mpl_backend(WARNING)
+
+    assert use_mock.call_count == 2
+    # Check that the most recent call to `matplotlib.use` has arg `Agg`
+    use_mock.assert_called_with("Agg")
+
+
+@pytest.mark.skipif(
+    not is_matplotlib_installed(),
+    reason="Test requires matplotlib to be installed.",
+)
+@patch("matplotlib.__version__", "0.0.0")
+def test_should_raise_import_error_for_version_check():
+    with pytest.raises(ImportError, match="A matplotlib version of at least"):
+        _set_mpl_backend(WARNING)
```

Comment:
As `_set_mpl_backend` is moved in `helpers.py`, I moved these tests to `test_helpers.py`.